pagopa/io-app-design-system
# [IOAPPX-333] Add dark mode support to all the `Module…` components

> [!caution]
> This PR depends on:
> * https://github.com/pagopa/io-app-design-system/pull/296

## Short description
This PR adds the dark mode support to all the `Module…` components

## List of changes proposed in this pull request
- Refactor code to apply theme keys to the `color` prop in the various typographic styles
- Refactor code to use `PressableModuleBase` in all the components
- Add the new `ModuleStatic` component to render the non-interactive version of the module. It can accept:
  - Two props: `startBlock` and `endBlock`
  - Single prop: `children` (standard mode)
- Update some examples included in the `Modules` screen to test all the possible variants

### Preview

https://github.com/pagopa/io-app-design-system/assets/1255491/fe8c1a16-5b15-4605-8877-92f46a5ed549



## How to test
1. Launch the example app
2. Check the **Modules** screen, both in light and dark mode
